include/nbl/core/containers/DoublyLinkedList.h

```diff
@@ -277,14 +287,14 @@ class DoublyLinkedList
 	}
 
 	// Iterator stuff
-	Iterator begin();
-	Iterator end();
-	Iterator cbegin() const;
-	Iterator cend() const;
-	std::reverse_iterator<Iterator> rbegin();
-	std::reverse_iterator<Iterator> rend();
-	std::reverse_iterator<Iterator> crbegin() const;
-	std::reverse_iterator<Iterator> crend() const;
+	Iterator<true> begin();
```

Comment:
Instead of <true> <false> use "Iterator" and "Const Iterator" names
make naming convention consistent with all other typdefs in the LRUCache and DoublyLinkedList, use `iterator` and `const_iterator` just like stl
your class name could still be `Iterator` templated.